app/src/main/java/com/example/groupstudyingapp/FireStoreHandler.java

```diff
@@ -100,21 +110,83 @@ public void onFailure(@NonNull Exception e) {
+     * @param newQuestion - the new question to which the new image belongs
+     */
+    public void uploadQuestionImage(final Uri localImagePath, String storedImagePath,
+                                    final Question newQuestion){
+        // Create a storage reference from our app
+        StorageReference storageRef = storage.getReference();
+        final StorageReference questionsRef = storageRef.child(storedImagePath);
+        UploadTask uploadTask = questionsRef.putFile(localImagePath);
+
+        // Register observers to listen for when the download is done or if it fails
+        uploadTask.addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception exception) {
+                Log.i(UNSUCCESSFUL_IMAGE_UPLOAD, "unsuccessful image upload");
+            }
+        }).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
             @Override
-            public void onFailure(@NonNull Exception e) {
-                Toast.makeText(context, "Error!", Toast.LENGTH_SHORT).show();
+            public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
+                Intent intent = new Intent();
+                intent.setAction(IMAGE_UPLOADED);
+                context.sendBroadcast(intent);
             }
         });
+        updateNewQuestionUri(newQuestion, questionsRef, uploadTask);
     }
 
+    /**
+     * Gets the URI of the recently uploaded image of the newQuestion
+     * @param newQuestion the newQuestion //todo - needs to be id and not object?
+     * @param questionsRef a reference to the questions images storage on firebase
+     * @param uploadTask the upload task that uploads the image
+     */
+    private void updateNewQuestionUri(final Question newQuestion,
+                                      final StorageReference questionsRef, UploadTask uploadTask) {
+        Task<Uri> urlTask = uploadTask.continueWithTask(new Continuation<UploadTask.TaskSnapshot,
+                Task<Uri>>() {
+            @Override
+            public Task<Uri> then(@NonNull Task<UploadTask.TaskSnapshot> task) throws Exception {
+                if (!task.isSuccessful()) {
+                    throw task.getException();
+                }
+                // Continue with the task to get the download URL
+                return questionsRef.getDownloadUrl();
+            }
+        }).addOnCompleteListener(new OnCompleteListener<Uri>() {
+            @Override
+            public void onComplete(@NonNull Task<Uri> task) {
+                if (task.isSuccessful()) {
+                    newQuestion.setLink(task.getResult());
+                    updateData(); // todo - empty method
+                    Log.i("URI", newQuestion.getLink().toString()); //todo delete
+                } else {
+                    // todo - handle failure
+                }
+            }
+        });
+    }
 
     public void buildDB() { // call once
```

Comment:
maybe we should make the DB on a separate local JSON file, so when we set the data we do it only once and iterate on the JSON lines (talk to me for further details).